### ISSUE TITLE
XFree86 multimedia key support

### ISSUE THREAD

**Issue by bryce-carson** [kmonad/kmonad#579]
Many keyboards have multimedia keys, such as the Microsoft Natural Ergonomic 4000.

I discovered that I am able to set custom keyboard shortcuts with these keys in GNOME Settings, but it would've been nice to manage these keys with KMonad along with all my other keys.

---

This is the information I found about these scan codes: https://wiki.linuxquestions.org/wiki/XF86_keyboard_symbols.

I recalled that these keys are sometimes reported as `XF86[...]` in the terminal, so I was able to find the link above using that keyword (XF86).

Thanks for the wonderful system software!

**Comment by MuhammedZakir** [comment]
It's already supported. You can find the names KMonad uses for those by passing `-l debug` cli flag. Note that, some(most?) keyboards show up as seperate event devices -- one for "normal" keys, one for consumer/multimedia keys, and one for power-control(?)/system keys. If that's the case for your keyboard, then you will have to run `kmonad` for each of them.

If your keyboard is shown up as 2 event devices, and if you want to get a complete picture when writing layers, then you can do something like:

1. use a layer like:
```
(defsrc
   srch  mail  hmpg  <other multimedia keys ...>
  esc  f1 <other keys>
  <rest of the keys>
)
```

2. and, in `input` in `defcfg`, use event device corresponding to "normal" keys (as is usually done).

3. Then, when running KMonad, run it twice:
```sh
# for normal keys
kmonad .config/kmonad/<your_config>

# for multimedia keys
kmonad .config/kmonad/<your_config> --input 'device-file "/dev/input/<mediakeys_event_device>"' --output 'KMonad: kb name: multimedia keys'
```

**Comment by david-janssen** [comment]
@bryce-carson: did you manage with Muhammed's advice? Can I close this issue?

**Comment by bryce-carson** [comment]
I haven't upgraded to the unreleased version, so when I tried using `sticky-key` (for fun, interest, and because I've been thinking about that anyways) it didn't work.

However, `-l debug` worked and I was able to identify what Kmonad interprets my multimedia keys as, so I will close the issue with this comment.

I do see the two devices, and I'll try a configuration with them, but I do belive all is good with this.
